### PR TITLE
Rewrite division and modulo with constant divisor

### PR DESCRIPTION
**Pull request by SirYwell**
yeah

### CODE REVIEW

**Review by I-Al-Istannen — COMMENTED** [review]
Looks alright but I am not going to understand the magic if that's alright with you @SirYwell. You want to force this or should I rubber stamp it?

File: src/main/java/com/github/firmwehr/gentle/backend/ir/visit/DjungelskogVisitor.java [hunk 1]
```diff
@@ -274,6 +276,28 @@ public String visit(IkeaDiv div) {
 		return result;
 	}
 
+	@Override
+	public String visit(IkeaShr shr) {
+		String result = "";
```
Comment (chrisliebaer) [inline src/main/java/com/github/firmwehr/gentle/backend/ir/visit/DjungelskogVisitor.java:281]: git didn't pick that up I guess, but code selection requires each visit method to check if the node has been replace, once you rebase on master. Just a heads up.

File: src/main/java/com/github/firmwehr/gentle/backend/ir/codegen/CodeSelection.java [hunk 2]
```diff
@@ -386,6 +390,22 @@ public void visit(Store node) {
 		block.nodes().add(ikeaMovStore);
 	}
 
+	@Override
+	public void visit(Shr node) {
+		IkeaBløck block = blocks.get((Block) node.getBlock());
```
Comment (chrisliebaer) [inline src/main/java/com/github/firmwehr/gentle/backend/ir/codegen/CodeSelection.java:395]: git didn't pick that up I guess, but code selection requires each visit method to check if the node has been replace, once you rebase on master. Just a heads up.

Comment (SirYwell) [reply src/main/java/com/github/firmwehr/gentle/backend/ir/codegen/CodeSelection.java:511]: rebased and updated, please check if everything is correct

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**